### PR TITLE
Update calculateAvailableKinds to support both string & object

### PR DESCRIPTION
**Pull request by benjaminapetersen**

### CODE REVIEW

File: dist/origin-web-common-services.js [hunk 1]
```diff
@@ -478,7 +483,7 @@ angular.module('openshiftCommonServices')
       _.each(api.resources.v1, function(resource) {
         if (resource.namespaced || includeClusterScoped) {
           // Exclude subresources and any rejected kinds
-          if (resource.name.indexOf("/") >= 0 || _.contains(rejectedKinds, resource.kind)) {
+          if (_.contains(resource.name, '/') || _.find(rejectedKinds, {kind: resource.kind})) {
```
Comment (jwforres) [inline dist/origin-web-common-services.js]: dont you need `group: ''` here

File: dist/origin-web-common-services.js [hunk 2]
```diff
@@ -495,7 +500,7 @@ angular.module('openshiftCommonServices')
       var preferredVersion = defaultVersion[group.name] || group.preferredVersion;
       _.each(group.versions[preferredVersion].resources, function(resource) {
         // Exclude subresources and any rejected kinds
-        if (resource.name.indexOf("/") >= 0 || _.contains(rejectedKinds, resource.kind)) {
+        if (_.contains(resource.name, '/') || _.find(rejectedKinds, {kind: resource.kind})) {
```
Comment (jwforres) [inline dist/origin-web-common-services.js]: you need group here, or its just going to do the same thing it does today

Comment (benjaminapetersen) [inline dist/origin-web-common-services.js]: Tinkering with this, atm it is greedy, dropping both `Binding` and `{ kind: "Binding", group: "servicecatalog.k8s.io" }`.

File: dist/origin-web-common.js [hunk 3]
```diff
@@ -896,44 +896,6 @@ angular.module('openshiftCommonUI')
       }
     };
   }]);
```
Comment (benjaminapetersen) [inline dist/origin-web-common.js]: Pulling this out into the `test/fixtures` dir.... was it supposed to end up in the dist?

Comment (jwforres) [inline dist/origin-web-common.js]: from the perspective of what should happen in the console, we never want OPENSHIFT_CONFIG being included in dist, that file should always be guaranteed to come from the generated config.js and nowhere else.  So removing it from common's dist is correct.  If that breaks the catalog app for some reason then we'll need to fix the catalog app.

Comment (benjaminapetersen) [inline dist/origin-web-common.js]: Yeah that seemed off.

**Comment by benjaminapetersen** [comment]
Ok, I think this is ready for review.  Added a test & pulled some things out into fixtures.

File: test/spec/fixtures/constants.js [hunk 4]
```diff
@@ -0,0 +1,12 @@
+window.OPENSHIFT_CONSTANTS = window.OPENSHIFT_CONSTANTS || {};
+
+window.OPENSHIFT_CONSTANTS.AVAILABLE_KINDS_BLACKLIST = [
+    // These are k8s kinds that are not supported in the current release of OpenShift
+    { kind: "Binding", group: "servicecatalog.k8s.io" },
```
Comment (jwforres) [inline test/spec/fixtures/constants.js]: Update the comment so its clear that its only being blacklisted for testing purposes, the comment as-is is definitely misleading, we won't be blacklisting the `servicecatalog.k8s.io` Binding in the console.

Comment (benjaminapetersen) [inline test/spec/fixtures/constants.js]: will do.

**Pull request merged**